### DISCUSSION TITLE
Proposal to support only the latest minor release

### DISCUSSION THREAD

**Discussion by gbartolini** [cloudnative-pg/cloudnative-pg#4258] [Maintainers room]
In the ["Supported release" page](https://cloudnative-pg.io/documentation/current/supported_releases/) we currently state:

> We support the latest two minor releases, and we aim to create a new minor release every two months.

We supported the latest two minor releases, given that we could release a minor version every two months. Unfortunately, we have not been able to respect this timeframe. 1.21 took 6 months. 1.23 is already 4 months on the go.

I propose to support only the latest version as a community and keep ~~the additional month~~ two additional months for the previous version.

**Comment by gbartolini** [comment]
Obviously this proposal assumes that we don't lower the guard on continuous delivery.

**Comment by sxd** [comment]
This indeed makes a lot of sense since keeping more than one version in opensource it kind of hard, and will help us to focus more on new features than keep really old versions

**Comment by armru** [comment]
Yes, it makes sense. There is only so much that we can do as an open-source project.

**Comment by fcanovai** [comment]
I believe we're making our life more difficult in multiple ways:

1. Small features are added to release branches. If we kept features to main, and backported only fixes there would be less backporting work.
2. We have multiple release branches supported
3. We don't release often, so backports piles up

I believe all these aspects are increasing the required amount of work, not just the second one.

Having said all that, I think that we could drop support for the second minor, but let's consider that users will only have one month to upgrade after a new minor release.

**Reply by gbartolini** [reply]
Shall we extend that period Francesco?

**Reply by fcanovai** [reply]
I think we could do a couple of months

**Comment by leonardoce** [comment]
I think we should keep on with the continuous delivery approach. I agree with supporting the latest minor version.
Let's give people two months of time to update.

**Comment by gbartolini** [comment]
Ok, so at the moment we are converging towards:

- the latest minor version
- extend the additional support to 2 months for the older version

**Comment by gbartolini** [comment]
@mnencia and @phisco thoughts?

**Comment by phisco** [comment]
Makes sense to me! We extensively test upgrades, so users should not be scared of upgrades, and it's a huge effort on us to support multiple versions, with little to no benefit. 👍

**Comment by mnencia** [comment]
I vote for reducing the number of versions. The best version of the operator is always the last one, and I do not see big advantages in maintaining multiple versions.

**Comment by smiyc** [comment]
I agree with only supporting the latest minor version but would vote for 3 additional month.

I have one maintenance window each month.
After a new release I upgrade the demo cluster asap (outside of the maintenance window).
In the 1st window I do test and in the 2nd prod.
So depending on the release date I could keep up with a supported prod environment, but if the release is shortly after my window, I can't.

**Reply by gbartolini** [reply]
I think 3 months is a good compromise.

**Comment by gbartolini** [comment]
Adopted in 1.23.0
